### PR TITLE
Fixes #11573: Sync plans available_products was missed during convers…

### PR DESCRIPTION
**Pull request by ehelms**
…ion.

This controller action was missed during scoped search conversion
and throws undefined mapping errors to users using this or the UI
page tied to it.

### CODE REVIEW

File: app/controllers/katello/api/v2/sync_plans_controller.rb [hunk 1]
```diff
@@ -87,18 +87,12 @@ def destroy
...
-
-      options = {
-        :filters       => filters,
-        :load_records? => true
-      }
-
-      products = item_search(Product, params, options)
-      respond_for_index(:collection => products)
+      options = {:includes => [:sync_plan, :provider], :resource_class => Product}
+      respond_for_index(:collection => scoped_search(products, :name, :desc, options), :template => '../products/index')
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/sync_plans_controller.rb]: What about just drop this action and use products#index?

**Comment by daviddavis** [comment]
Your git commit message is messy. [Check out this link](http://chris.beams.io/posts/git-commit/).

File: app/controllers/katello/api/v2/sync_plans_controller.rb (hunk 1 above)
Comment (ehelms) [inline app/controllers/katello/api/v2/sync_plans_controller.rb]: I could to that, but I would have to either:

a) introduce checking for 'available_products' in the URL or action and do some special logic
b) Introduce the available_for concept (https://github.com/Katello/katello/blob/b5f0c408706055e82c04bdd79eef096aee8e84d0/app/controllers/katello/api/v2/repositories_controller.rb#L47) and remove this as an API endpoint through a deprecation cycle

Comment (daviddavis) [inline app/controllers/katello/api/v2/sync_plans_controller.rb]: Another option:

c) Do both option (b) and option (a) until the route is deprecated and then remove option (a)

I'll leave it up to you. I have a slight preference for b or c but would be fine with a.

**Comment by ehelms** [comment]
Closing in favor of https://github.com/Katello/katello/pull/5450

**Pull request closed without merging**